lab-lee/lib/storage.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const Promise = require('bluebird');
+const createError = require('http-errors');
+const debug = require('debug')('duck:storage');
+const fs = Promise.promisifyAll(require('fs'), {suffix: 'Prom'});
+
+const mkdirp = Promise.promisifyAll(require('mkdirp'), {suffix: 'Prom'});
+
+module.exports = exports = {};
+
+exports.createItem = function(schemaName, item) {
+  debug('createItem');
+
+  if (!schemaName) return Promise.reject(createError(400, 'expected schemaName'));
+  if (!item) return Promise.reject(createError(400, 'expected an item'));
+
+  let json = JSON.stringify(item);
+  return fs.accessProm(`${__dirname}/../data/${schemaName}`)
+  .catch(err => {
+    if(err.code === 'ENOENT') {
+      return mkdirp.mkdirpProm(`${__dirname}/../data/${schemaName}`);
+    }
+    return Promise.reject(err);
+  })
+  .then( () =>
+  fs.writeFileProm(`${__dirname}/../data/${schemaName}/${item.id}.json`, json)
+)
+  .then( () => item)
+  .catch( err => Promise.reject(createError(500, err.message)));
+};
+
+exports.putItem = function(schemaName, item) {
```

Comment:
So, you shouldn't need these method if you refactor the way that Duncan showed in class today... which it probably a better way of doing it :) You'll pretty much be making use of fetchItem and createItem inside of your updateDuck method.